src/emq_web_hook.erl

```diff
@@ -225,15 +225,40 @@ on_message_acked(ClientId, Username, Message = #mqtt_message{topic = Topic}, {Fi
 %% Internal functions
 %%--------------------------------------------------------------------
 
+-spec get_timestamp_ms() -> integer().
+get_timestamp_ms() ->
+    {Mega, Sec, Micro} = os:timestamp(),
+    (Mega*1000000 + Sec)*1000 + round(Micro/1000).
+
 send_http_request(Params) ->
     Params1 = iolist_to_binary(mochijson2:encode(Params)),
     Url = application:get_env(?APP, url, "http://127.0.0.1"),
     ?LOG(debug, "Url:~p, params:~s", [Url, Params1]),
-    case httpc:request(post, {Url, [], "application/json", Params1}, [{timeout, 5000}], []) of
-        {ok, _} -> 
+    case perform_request(post, {Url, [], "application/json", Params1}, [{timeout, 5000}], [], 1, 4) of
+        ok ->
+            ok;
+        {error, Reason} ->
+            ?LOG(error, "HTTP request error: ~p", [Reason]),
+            ok
+    end.
+
+perform_request(Method, Req, HTTPOpts, Opts, Attempt, MaxAttempt) ->
+    StartTime = get_timestamp_ms(),
+    case httpc:request(Method, Req, HTTPOpts, Opts) of
+        {ok, _} ->
+            Tags = #{ pod_name => os:getenv("HOSTNAME"), status => "success", attempt => Attempt },
+            dogstatsd:histogram("mqtt_broker.webhook.latency_ms", get_timestamp_ms() - StartTime, Tags),
             ok;
         {error, Reason} ->
-            ?LOG(error, "HTTP request error: ~p", [Reason]), ok %% TODO: return ok?
+            Tags = #{ pod_name => os:getenv("HOSTNAME"), status => "failure", attempt => Attempt },
+            dogstatsd:histogram("mqtt_broker.webhook.latency_ms", get_timestamp_ms() - StartTime, Tags),
+            case Attempt + 1 =< MaxAttempt of
+                true ->
+                    timer:sleep(trunc(math:pow(10, Attempt - 1))),
```

Comment:
Please, confirm this will not block the application in any way

Reply:
Good question, I'm really not sure if this sleep() will block the application, but I "copied" the code from the original authors of a more recent version of the plugin, look:

https://github.com/emqx/emqx-web-hook/blob/master/src/emqx_web_hook.erl#L289

Reply:
We decided to not implement the retries right now!